qa/src/test/java/io/camunda/migrator/qa/history/entity/HistoryProcessInstanceTest.java

```diff
@@ -0,0 +1,186 @@
+    for (ProcessInstanceEntity processInstance : processInstances) {
+      // and process instance has expected state
+      assertC8ProcessInstanceFields(processInstance, historicProcessInstance, "incidentProcessId",
+          ProcessInstanceEntity.ProcessInstanceState.ACTIVE, null, null, false, true);
+    }
+  }
+
+  @Test
+  @Disabled("https://github.com/camunda/camunda-bpm-platform/issues/5359")
+  public void shouldCheckCalledProcessParentElementKey() {
+
+    // given
+    deployer.deployCamunda7Process("callActivityProcess.bpmn");
+    deployer.deployCamunda7Process("calledActivitySubprocess.bpmn");
+    ProcessInstance parentInstance = runtimeService.startProcessInstanceByKey("callingProcessId");
+    ProcessInstance subInstance = runtimeService.createProcessInstanceQuery()
+        .superProcessInstanceId(parentInstance.getProcessInstanceId())
+        .singleResult();
+    completeAllUserTasksWithDefaultUserTaskId();
+    HistoricActivityInstance callActivity = historyService.createHistoricActivityInstanceQuery()
+        .activityId("callActivityId")
+        .processInstanceId(subInstance.getId())
+        .singleResult();
+    dbClient.insert(callActivity.getId(), null, IdKeyMapper.TYPE.HISTORY_PROCESS_DEFINITION);
+    // when
+    historyMigrator.migrate();
+
+    // then
+    assertThat(searchHistoricProcessInstances("calledProcessInstanceId")).isEmpty();
+  }
+
+  protected static void assertC8ProcessInstanceFields(ProcessInstanceEntity processInstance,
+                                                      HistoricProcessInstance historicProcessInstance,
+                                                      String processDefinitionId,
+                                                      ProcessInstanceEntity.ProcessInstanceState processInstanceState,
+                                                      String versionTag,
+                                                      String tenantId,
+                                                      boolean hasParent,
+                                                      boolean hasIncidents) {
+    assertThat(processInstance.processDefinitionId()).isEqualTo(processDefinitionId);
+    assertThat(processInstance.state()).isEqualTo(processInstanceState);
+    assertThat(processInstance.processInstanceKey()).isNotNull();
+    assertThat(processInstance.processDefinitionKey()).isNotNull();
+    assertThat(processInstance.tenantId()).isEqualTo(tenantId);
+    assertThat(processInstance.startDate()).isEqualTo(
+        ConverterUtil.convertDate(historicProcessInstance.getStartTime()));
+    assertThat(processInstance.endDate()).isEqualTo(ConverterUtil.convertDate(historicProcessInstance.getEndTime()));
+    assertThat(processInstance.processDefinitionVersion()).isEqualTo(1);
+    assertThat(processInstance.processDefinitionVersionTag()).isEqualTo(versionTag);
+
+    if (hasParent) {
+      assertThat(processInstance.parentProcessInstanceKey()).isNotNull();
```

Comment:
Same question here re. asserting the real key value

Reply:
I will adjust assert.